source/core/TDCodecVia.cpp

```diff
@@ -1790,6 +1786,7 @@ void TDViaParser::ParseClump(VIClump* viClump, InstructionAllocator* cia)
             if (!instruction) {
                 LOG_EVENTV(kSoftDataError, "Instruction not generated '%.*s'", FMT_LEN_BEGIN(&instructionNameToken));
             }
+            argExpressionTokens.Clear();
```

Comment:
You should have a destructor that does this, and rely on that rather than doing Clear explicitly.
Otherwise, incorrect usage could leak.

Reply:
I will remove this because I don't think it's necessary and will result in more allocations instead of less. The idea wasn't to free memory before destruction, but to free it before we process the next instruction. The object doesn't get destructed in between instructions.

As far as leaks go, shouldn't the std::vector be deallocated automatically since it is a non-pointer member variable of the class?

Reply:
Yeah, that's true.  Sorry, ignore this comment.